src/providers/vmware/amd64.rs

```diff
@@ -24,11 +29,24 @@ impl VmwareProvider {
             vmw_backdoor::probe_backdoor()
         })?;
 
-        let mut erpc = backdoor.open_enhanced_chan()?;
+        let mut erpc = vmw_backdoor::EnhancedChan::open(&mut backdoor)?;
         let guestinfo_net_kargs = Self::fetch_guestinfo(&mut erpc, INITRD_NET_KARGS)?;
 
+        drop(erpc);
```

Comment:
Multiple calls on a channel don't work:
```
    2: vmware backdoor error: erpc send_command_len failed
```
and the channel holds the mutable reference, hence to create a new channel, I needed to drop. I think the explicit call is better than doing this implicitly through blocks because this will be confusing for the next person to touch.

Reply:
Hmm, possibly there's a bug there in `vmw_backdoor` that we need to fix?

> I think the explicit call is better than doing this implicitly through blocks because this will be confusing for the next person to touch.

But blocks have the advantage that the compiler guarantees that the dropping is implemented correctly rather than having to verify the manual implementation. A comment on the first instance should suffice to explain to the reader what's going on.

Reply:
Ok, have changed it now to use blocks.